docs/developers/ui-components.rst

```diff
@@ -0,0 +1,301 @@
+results in multiple such elements on a page, additional work is needed to give them
+distinctive accessible labels.
+
+The question "How would this appear to a screenreader" should always be at the forefront
+when writing HTML.
+
+Block-Element-Modifier (BEM)
+----------------------------
+
+`BEM <https://getbem.com/>`_ is a methodology for reusable components.
+
+**Blocks**
+
+For the CSS and CSS class names, the BEM methodology is used. In a nutshell, this means
+the outer element gets a ``.my-block`` classname that encapsulates the component and
+is the outer CSS target selector.
+
+Example:
+
+.. code-block:: html
+
+    <footer class="openforms-footer">
+        ...
+    </footer>
+
+.. tip::
+
+    * You should try to limit the number of different blocks that exist in a codebase.
+    * Avoid using extremely specific blocks that cannot be reused in other places.
+
+**Elements**
+
+The element can contain zero or more child elements, which get a ``.my-block__element-1``
+or ``.my-block__element-2`` class name. They can exist anywhere in the HTML tree, they
+are not limited to direct children.
+
+A child element cannot exist / does not have meaning without its block parent.
+
+For example:
+
+.. code-block:: html
+
+    <footer class="openforms-footer">
+        <ul class="openforms-footer__group">
+            <li class="openforms-footer__item">
+                <a href="#" class="utrecht-link">Some link</a>
+            </li>
+        </ul>
+
+        <ul class="openforms-footer__group"> ... </ul>
+
+        <ul class="openforms-footer__group"> ... </ul>
+    </footer>
+
+.. warning::
+
+    You can have only **one** occurrence of the `__` separator. A class name like
+    ``.block__element__nested`` is not valid according to BEM, and a red flag that
+    your class names are mimicking your HTML structure.
+
+.. tip::
+
+    It's okay to combine an element and block class name, e.g.
+    ``utrecht-link openforms-footer__link`` can make sense. The styles from the element
+    are then scoped to when a link is a child of the block element.
+
+**Modifiers**
+
+Modifiers indicate variants of a block and/or element. The double dash (``--``)
+separator is used: ``.block--modifier`` and ``.block__element--modifier``.
+
+Modifiers can be applied to both blocks and elements, e.g.:
+
+.. code-block:: html
+
+    <footer class="openforms-footer openforms-footer--vertical">
+        <ul class="openforms-footer__group openforms-footer__group--compact">
+            ...
+        </ul>
+    </footer>
+
+.. warning::
+    A modifier is always relative to its base block or element, meaning you must include
+    that base class name too. ``<a href="#" class="utrecht-link--html-a">...</a>`` is
+    incorrect usage, instead it must be:
+    ``<a href="#" class="utrecht-link utrecht-link--html-a">...</a>``
+
+**SCSS helpers**
+
+The library ``microscope-sass`` has a helper to build the correct BEM class names.
```

Comment:
I am again confused with this library, I thought we wanted to gradually get rid of it.